src/Compilers/CSharp/Portable/Symbols/TypeWithAnnotations.cs

```diff
@@ -479,6 +483,10 @@ internal TypeWithAnnotations SubstituteTypeCore(AbstractTypeMap typeMap)
             {
                 newAnnotation = NullableAnnotation;
             }
+            else if (newTypeWithModifiers.NullableAnnotation == NullableAnnotation.Ignored)
```

Comment:
>newTypeWithModifiers.NullableAnnotation == NullableAnnotation.Ignore [](start = 21, length = 68)

It looks like we already have a special handling for indexed type parameters that has the same effect. If that is the case, I think we should reuse this mechanism rather going through the trouble of dealing with the new annotation. We might need to use ConstructedFrom in diagnostics in affected components though.

Reply:
We can't use indexed type parameters because (1) they don't have names, and (2) they don't have constraints (so the logic in `AreParameterAnnotationsCompatible` which simulates an assignment produces wrong values/states in `GetParameterState`).
We could fix the first problem by introducing a new type parameter symbol. But the second problem is tougher, and I don't see a good way to solve it. 
So using a new annotation seems a better fit to solve this problem.

Reply:
> We can't use indexed type parameters because (1) they don't have names, and (2) they don't have constraints (so the logic in AreParameterAnnotationsCompatible which simulates an assignment produces wrong values/states in GetParameterState).

I would like to get a better understanding of these issues. Let's sync offline

---
In reply to: [535501474](https://github.com/dotnet/roslyn/pull/49723#discussion_r535501474) [](ancestors = 535501474)

Reply:
Would it be possible to remove the special handling for indexed type parameters and use the new annotation instead? 

---
In reply to: [535509257](https://github.com/dotnet/roslyn/pull/49723#discussion_r535509257) [](ancestors = 535509257,535501474)

Reply:
> Would it be possible to remove the special handling for indexed type parameters and use the new annotation instead?

Interesting idea. Thanks. Pushed a commit for that change